csrc/fusion.cpp

```diff
@@ -592,12 +584,7 @@ void Fusion::resetTvUses() {
   // Same as in register expr
   for (auto expr : used_exprs) {
     for (Val* input : expr->inputs()) {
-      auto uses_copy = input->uses();
-      if (std::find(uses_copy.begin(), uses_copy.end(), expr) ==
-          uses_copy.end()) {
-        uses_copy.push_back(expr);
-        input->setUses(uses_copy);
-      }
+      input->addUse(expr);
```

Comment:
I'm not sure when this change happened, but `addUse` for tensors doesn't seem to do anything other than invalidating the use info. So, this doesn't seem to set any use of tensors, does it?

Reply:
Indeed I overlooked that. Tried addressing it by adding a flag to addUse to enable forcing the insertion. Only used during resetTvUses